main.tf

```
@@ -0,0 +1,73 @@
+/**
+ * # AWS EKS Argo Workflows Terraform module 
+ *
+ * A Terraform module to deploy the [Argo Workflows](https://argoproj.github.io/workflows/) on Amazon EKS cluster.
+ *
+ * [![Terraform validate](https://github.com/lablabs/terraform-aws-eks-argo-workflows/actions/workflows/validate.yaml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-argo-workflows/actions/workflows/validate.yaml)
+ * [![pre-commit](https://github.com/lablabs/terraform-aws-argo-workflows/actions/workflows/pre-commit.yml/badge.svg)](https://github.com/lablabs/terraform-aws-eks-argo-workflows/actions/workflows/pre-commit.yml)
+ */
+
+locals {
+  addon = {
+    name = "argo-workflows" 
+    namespace = "argo-workflows" 
+
+    helm_chart_version = "0.20.8"
+    helm_repo_url      = "https://argoproj.github.io/argo-helm"
+  }
+
+  addon_irsa = {
+    "${local.addon.name}-server" = {
+      service_account_name = "${local.addon.name}-server"
+      irsa_role_create = var.server_irsa_role_create != null ? var.server_irsa_role_create : true
+      irsa_role_name = "${local.addon.name}-serv-irsa" # Max length of rolename was reached, using shortened version
+      irsa_additional_policies = var.server_irsa_additional_policies != {} ? var.server_irsa_additional_policies : tomap({})
+    }
+    "${local.addon.name}-controller" = {
+      service_account_name = "${local.addon.name}-controller"
+      irsa_role_create = var.controller_irsa_role_create != null ? var.controller_irsa_role_create : true
+      irsa_role_name= "${local.addon.name}-ctrl-irsa" # Max length of rolename was reached, using shortened version
+      irsa_additional_policies = var.controller_irsa_additional_policies != {} ? var.controller_irsa_additional_policies : tomap({})
+    }
+     "${local.addon.name}-workflow" = {
+      service_account_name = "${local.addon.name}-workflow"
+      irsa_role_create = var.workflow_irsa_role_create != null ? var.workflow_irsa_role_create : false
+      irsa_role_name= "${local.addon.name}-work-irsa" # Max length of rolename was reached, using shortened version
```

```suggestion
      irsa_role_name= "workflow" # Max length of rolename was reached, using shortened version
```
This should create a role with name `argo-workflows-irsa-workflow` which might be it

---

Fixed

---

In module addon-irsa -> iam.tf -> locals 
```
 irsa_role_name           = trim("${var.irsa_role_name_prefix}-${var.irsa_role_name}", "-")
```

This is causing that we use prefix and role_name concated

Default for as per addon-irsa.tf"
```
 irsa_role_name           = trim("${var.irsa_role_name_prefix}-${var.irsa_role_name}", "-")
```

This is causing longer name for default naming for roles

I see following end result via AWS Console: `ref-dev-ew1-e-service-argo-workflows-argo-workflows-serv-irsa`

Thats why we hit the limitation of characters

Do you want me to change it somehow still ?